doc/developer_guide/index.md

```diff
@@ -2,6 +2,10 @@
 
 The Panel library is a complex project which provides a wide range of data interfaces and an extensible set of plotting backends, which means the development and testing process involves a wide set of libraries.
 
+This guide describes how to install and configure the development environment fully.
+
+If you have any problems with the steps here, please reach out in the `dev` channel on [Discord](https://discord.gg/rb6gPXbdAr).
+
 ## Preliminaries
```

Comment:
The organization doesn't really make sense anymore, since we now have a Preliminaries section with just one subsections. I'd probably move the pip and conda sections up under preliminaries but am open to other suggestions.

Reply:
I've reorganized as you suggested